File: src/main/java/org/digma/intellij/plugin/jaegerui/model/SpansWithResolvedLocationMessage.java
package org.digma.intellij.plugin.jaegerui.model;


import com.fasterxml.jackson.annotation.*;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;

import java.util.Map;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
@JsonSerialize
public record SpansWithResolvedLocationMessage(@JsonProperty(value = "type",required = true) String type,
                                               @JsonProperty(value = "action",required = true)  String action,
                                               @JsonProperty(value = "payload",required = true) Map<String,Importance> payload ) {

    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
    public SpansWithResolvedLocationMessage {
        //nothing to do , for jackson only
    }
}

File: src/main/kotlin/org/digma/intellij/plugin/ui/jaegerui/model/SpansWithResolvedLocationMessage.kt
package org.digma.intellij.plugin.ui.jaegerui.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.beans.ConstructorProperties


@JsonIgnoreProperties(ignoreUnknown = true)
data class SpansWithResolvedLocationMessage
@JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
@ConstructorProperties("type", "action", "payload")
constructor(
    val type: String,
    val action: String,
    val payload: Map<String, SpanData>
)